discovery-provider/src/eth_indexing/event_scanner.py

```diff
@@ -392,7 +388,8 @@ def _retry_web3_call(  # type: ignore
                 time.sleep(delay)
                 continue
             logger.warning("event_scanner.py | Out of retries")
-            raise
+            raise e
+    raise
```

Comment:
this should be removed?

Reply:
revise this

Reply:
Took your suggestion about restructuring code